pkgs/tools/X11/xdg-utils/default.nix

```
@@ -15,73 +16,233 @@ let
+  commonPrologue = "${writeText "xdg-utils-prologue" ''
+    export PATH=$PATH:${coreutils}/bin
+  ''}";
+
+  solutions = [
+    {
+      scripts = [ "bin/xdg-desktop-icon" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps;
+      fake.external = commonFakes ++ [
+        "gconftool-2"
+        "xdg-user-dir"
+      ];
+      keep."$KDE_SESSION_VERSION" = true;
+      prologue = commonPrologue;
+    }
+
+    {
+      scripts = [ "bin/xdg-desktop-menu" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps ++ [ gawk ];
+      fake.external = commonFakes;
+      keep."$KDE_SESSION_VERSION" = true;
+      prologue = commonPrologue;
+    }
+
+    {
+      scripts = [ "bin/xdg-email" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps ++ [ gawk "${placeholder "out"}/bin" ];
+      execer = [
+        "cannot:${placeholder "out"}/bin/xdg-mime"
+        "cannot:${placeholder "out"}/bin/xdg-open"
+      ];
+      fake.external = commonFakes ++ [
+        "exo-open"
+        "gconftool-2"
+        "gdbus"
+        "gio"
+        "gnome-open"
+        "gvfs-open"
+        "qtxdg-mat"
+        "xdg-email-hook.sh"
+      ];
+      fix."/bin/echo" = true;
+      keep = {
+        "$command" = true;
+        "$kreadconfig" = true;
+        "$THUNDERBIRD" = true;
+        "$utf8" = true;
+      };
+    }
+
+    {
+      scripts = [ "bin/xdg-icon-resource" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps;
+      fake.external = commonFakes;
+      keep."$KDE_SESSION_VERSION" = true;
+      prologue = commonPrologue;
+    }
+
+    {
+      scripts = [ "bin/xdg-mime" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps ++ [ file gawk ];
+      fake.external = commonFakes ++ [
+        "gio"
+        "gnomevfs-info"
+        "gvfs-info"
+        "kde4-config"
+        "kfile"
+        "kmimetypefinder"
+        "kmimetypefinder5"
+        "ktraderclient"
+        "ktradertest"
+        "mimetype"
+        "qtpaths"
+        "qtxdg-mat"
+      ];
+      fix."/usr/bin/file" = true;
+      keep = {
+        "$KDE_SESSION_VERSION" = true;
+        "$KTRADER" = true;
+      };
+    }
+
+    {
+      scripts = [ "bin/xdg-open" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps ++ [ nettools "${placeholder "out"}/bin" ];
+      execer = [
+        "cannot:${placeholder "out"}/bin/xdg-mime"
+      ];
+      fake.external = commonFakes ++ [
+        "cygstart"
+        "dde-open"
+        "enlightenment_open"
+        "env"
+        "exo-open"
+        "gdbus"
+        "gio"
+        "gnome-open"
+        "gvfs-open"
+        "kde-open"
+        "kfmclient"
+        "mate-open"
+        "mimeopen"
+        "open"
+        "pcmanfm"
+        "qtxdg-mat"
+        "run-mailcap"
+        "rundll32.exe"
+        "wslpath"
+      ];
+      keep = {
+        "$browser" = true;
+        "$KDE_SESSION_VERSION" = true;
+        "$printf" = true;
+      };
```

Upstream's source uses this `$printf` variable to invoke `/usr/bin/printf` if it's available, and otherwise use the shell builtin. Since we're controlling the interpreter, I think we can just rewrite this to plain `printf` and let resholve pick up the builtin.

```suggestion
      fix."$printf" = [ "printf" ];
      keep = {
        "$browser" = true;
        "$KDE_SESSION_VERSION" = true;
      };
```

For posterity: what they're doing here strikes me as strange and the commit that introduced it doesn't explain. I'm just projecting--but the only sense I can make of it is that they got bit by some subtle printf differences when users try running this with different shells, and figured they'd mitigate that by specifying a system path? Maybe they're imaging it would usually be a GNU printf and thus ~stable across shells? (But of course on macOS it'll be the BSD printf. Maybe BSD and GNU printf are more-compatible than different shell builtins?)

---

Applied.